### ISSUE TITLE
Internationalization (i18n) merging already translated files

### ISSUE THREAD

**Issue by mestefanivanov** [angular/angular#37655]
# 🚀 feature request regarding i18n

### Description
Hi all, are you considering some approach regarding i18n translations maintainability. What I mean is:
- what is the best approach when you already have a translated app and for example delete some feature and add new ones to the project. Are you planning some kind of merging between the already translated files and the new incoming sources.

### Solution I'd like
Command which overrides our current translated files which are described in the "angular.json". With the merge it can
- delete unused sources by id
- add the new sources

**Comment by petebacondarwin** [comment]
Thanks for suggesting this feature.

Currently we don't have enough capacity to do work on additional tooling around merging and cleaning up translation files. This is an area that could be dealt with by 3rd party tooling and at the moment I would put it as outside the scope of the framework. I believe that tools like POEditor can handle this sort of merging?

**Comment by destus90** [comment]
The similar discussion was here https://github.com/angular/angular-cli/issues/6552 with the official answer https://github.com/angular/angular-cli/issues/6552#issuecomment-316038368

**Comment by petebacondarwin** [comment]
You could also take a look at xliffmerge: https://github.com/martinroob/ngx-i18nsupport/tree/master/projects/xliffmerge

**Comment by julmot** [comment]
@petebacondarwin xliffmerge isn't compatible with Angular v11 anymore. It has previously been working even if the peerDependency was Angular v8. See https://github.com/martinroob/ngx-i18nsupport/issues/167

**Comment by tommyblue** [comment]
I'm adding i18n support to an angular app that will still be maintained and expanded after the 1st release. Having a stable solution to merge new strings to existing translations files would be really important

**Comment by jakob-bebop** [comment]
This is something we need as well!

**Comment by julmot** [comment]
I'm currently using Poeditor.com. It's working great, but it feels kinda odd because I have to pay monthly for just merging files. I don't need any of the contribution features of Poeditor.com.

**Comment by tommyblue** [comment]
For whoever is facing the same problems, we're now using https://github.com/martinroob/ngx-i18nsupport/tree/master/projects/xliffmerge and it works very well

**Comment by julmot** [comment]
@tommyblue Can't confirm this, as mentioned earlier it's not compatible with the most recent Angular version anymore. What Angular version are you using?

**Comment by tommyblue** [comment]
> @tommyblue Can't confirm this, as mentioned earlier it's not compatible with the most recent Angular version anymore. What Angular version are you using?

@julmot 11

**Comment by jakob-bebop** [comment]
I updated to angular 12, and my translation workflow with xliffmerge is working again.

**Comment by SyntaxErrorOnLine1** [comment]
I have xliffmerge working in Angular 12.

Install:
`npm install -D @ngx-i18nsupport/tooling --legacy-peer-deps`

Run:
`ng extract-i18n && ng run <project-name>:xliffmerge`

[Credit](https://medium.com/dailyjs/maintaining-multi-language-angular-applications-26b74df8d085)

I would really like to see this feature added to the CLI, though. Xliffmerge is only going to keep running for so long now the project is abandoned and i18n is simply not implemented properly in Angular until there is a way to maintain existing translations.

**Comment by Andrew-Marks-Trisept** [comment]
Agreed, i18n doesn't seem like a complete solution without the ability to maintain translations going forward.  It's like we are expected to translate our apps once, at the end, and never do it again.  Maybe if we were shipping boxed software that will never get updated... 

I'm really thankful for xliffmerge but worry it will stop working.

**Comment by faan11** [comment]
are there any updates about this feature? I also think that merging translation is a quite common demand in a generic website. Install @ngx-i18nsupport/tooling by avoiding dependencies check is a trick which also brings trouble after.

**Comment by petebacondarwin** [comment]
None at this stage. Since such a tool is not a trivial undertaking and would then require ongoing maintenance the decision to incorporate this into the framework itself has to be taken carefully. I will highlight this item at our next feature triage meeting to decide next steps.
